TOY-25 카메라 초점기능 구현 작업(TOY-15) 후 MVVM 패턴 적용 및 Kotlin 변환

pby2017 opened the pull request:
코드 리뷰 부탁드립니다.

1. MVVM 패턴 적용 / DataBinding 추가 / Kotlin 변환
* DataBinding을 사용한 MVVM 패턴을 적용했습니다.
* 기존 Java 코드를 Kotlin으로 변환했습니다.

2. dispose() 위치 변경
* TOY-15 PR 당시 반영되지 않은 `ResultHolder.dispose() 위치 변경 작업`을 수행했습니다.

pby2017 committed:
TOY-25 Kotlin 변환, databinding 추가, MVVM 구조로 분리

pby2017 committed:
TOY-25 ResultHolder 의 Bitmap dispose() 호출 위치 변경

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraViewModel.kt:
```diff
@@ -0,0 +1,100 @@
+import kr.co.yogiyo.rookiephotoapp.R
+
+class CameraViewModel(context: Context) : BaseObservable() {
+    private val DELAY_DURATIONS = mutableListOf(0, 2, 5, 10)
```
companion object로 빼는 편이 나을 거 같아요.

pby2017 replied:
네 반영하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraActivity.kt:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    private fun initCameraView() {
+        camera.addCameraListener(object : CameraListener() {
```
다음과 같이 run 블락을 사용하면 코드를 좀더 줄일 수 있습니다.

        camera.run {
            addCameraListener(object : CameraListener() {
                override fun onPictureTaken(jpeg: ByteArray?) {
                    if (jpeg != null) {
                        CameraUtils.decodeBitmap(jpeg, CameraUtils.BitmapCallback { bitmap ->
                            if (bitmap == null) {
                                finish()
                                return@BitmapCallback
                            }
                            ResultHolder.setBitmap(bitmap)
                            val intent = Intent(this@CameraActivity, PreviewActivity::class.java)
                            startActivity(intent)
                        })
                    }
                }
            })
            mapGesture(Gesture.TAP, GestureAction.FOCUS_WITH_MARKER)

            val ratio = SizeSelectors.aspectRatio(AspectRatio.of(3, 4), 0f)
            val result = SizeSelectors.or(ratio, SizeSelectors.biggest())
            setPictureSize(result)
        }

pby2017 replied:
네 반영하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraViewModel.kt:
```diff
@@ -0,0 +1,100 @@
+
+        } else if (captureTimer > 0) {
+            delayMessageLabel.set(captureTimer)
+            Log.d(CameraViewModel::class.java.simpleName, "$captureTimer")
```
TAG도 보통 companion object에 선언해서 사용하는 편이에요.

pby2017 replied:
네 반영하겠습니다.

bonchangkoo reviewed (approved):
코드 확인했습니다.

pby2017 committed:
TOY-25 코드리뷰 반영 (companion object, kotlin run, 중복제거)

* Camera 액티비티 kotlin run 코드 삽입
* Camera 액티비티 파일 중복 제거
* Preview 액티비티 onDestroy 중복 제거
* Camera 뷰모델 static 값 companion object 블록으로 이동

pby2017 committed:
TOY-25 병합 충돌 작업 수행 (develop 브랜치)

Merge branch 'develop' into TOY-25-camera-mvvm-kotlin

# Conflicts:
#	app/build.gradle

bonchangkoo commented:
코드 확인했습니다.

hyejin830 reviewed (approved):
코드 확인했습니다.
